skimage/transform/_geometric.py

```diff
@@ -1092,52 +1255,63 @@ class SimilarityTransform(EuclideanTransform):
 
     Parameters
     ----------
-    matrix : (3, 3) array, optional
+    matrix : (dim+1, dim+1) array, optional
         Homogeneous transformation matrix.
     scale : float, optional
-        Scale factor.
+        Scale factor. Implemented only for 2D and 3D.
     rotation : float, optional
         Rotation angle in counter-clockwise direction as radians.
-    translation : (tx, ty) as array, list or tuple, optional
-        x, y translation parameters.
+        Implemented only for 2D and 3D. For 3D, this is given in XZX Euler
+        angles.
```

Comment:
XZX is not right here is it? It looks like we use axes in the order 0, 1, 2 in _euler_matrix which should be XYZ (assuming axis=0 means X?). XZX is apparently one of 12 possible conventions, but it didn't seem to be the one used here.

Another form for 3D that I find fairly intuitive is where the user specifies a unit vector corresponding to the [axis of rotation and then a single rotation angle about that vector. There is an [explicit formula for that form](https://en.wikipedia.org/wiki/Rotation_matrix#Rotation_matrix_from_axis_and_angle).

Reply:
> XZX is not right here is it?

Ah, yes, XZX is left over from an older version of the code. Will fix.

> Another form for 3D that I find fairly intuitive is where the user specifies a unit vector corresponding to the axis of rotation and then a single rotation angle about that vector.

Could change to that if people think that's better...

Reply:
@grlee77 since the 3D Euler angles and the 3D vector + rotation are unambiguously distinguishable, I'm going to leave it as 3D Euler angles for now (fixed the docstring), and we can revisit in the future if someone clamors for alternate parameterizations.